### PR TITLE
DApp3 Search bar redesign

### LINKED ISSUE #2410: Browse Suggested Categories
Once the search field is focused, reveal suggested categories to browse.

Mobile: https://zpl.io/aRxQe1r
Desktop: https://zpl.io/bJpYe7D

We will manually feature a handful of categories with icons to be determined.

**Bonus**: allow for filtering with a combination of category selection and search query.

### PR DESCRIPTION
**Pull request by shahthepro**
Should close #2410 

Have to update the cancel icon and the search icon. This shouldn't block the merge though.

### CODE REVIEW

File: dapps/marketplace/src/pages/listings/Listings.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ class Listings extends Component {
         <DocumentTitle pageTitle={<fbt desc="listings.title">Listings</fbt>} />
         <div className="container listings-container">
           {this.props.isMobile ? (
-            <Search className="search" placeholder />
+            <Search className="search" placeholder showDropdown />
```
Comment (nick) [inline dapps/marketplace/src/pages/listings/Listings.js]: are there any instances of the Search component that don't need showDropdown?

File: dapps/marketplace/src/pages/listings/_Search.js [hunk 2]
```diff
@@ -7,49 +7,145 @@ import { fbt } from 'fbt-runtime'
 import { getStateFromQuery } from './_filters'
 
 import withConfig from 'hoc/withConfig'
+import withIsMobile from 'hoc/withIsMobile'
 
 class Search extends Component {
```
Comment (nick) [inline dapps/marketplace/src/pages/listings/_Search.js]: wondering if we can turn this into a functional component. No worries if not... in general trying to take the opportunity to convert to functional when working on those parts of the code, but only if it's not a big pain.

File: dapps/marketplace/src/pages/listings/_Search.js [hunk 3]
```diff
@@ -7,49 +7,145 @@ import { fbt } from 'fbt-runtime'
...
+
+    return (
+      // tabIndex to keep the focus on click
+      <div
+        className={`search-dropdown${isMobile ? '' : ' floating'}`}
+        tabIndex={isMobile ? -1 : 0}
+      >
+        {!isMobile && (
+          <div className="title">
+            <fbt desc="Search.Categories">CATEGORIES</fbt>
```
Comment (nick) [inline dapps/marketplace/src/pages/listings/_Search.js]: Can we convert to uppercase in CSS

Comment (shahthepro) [reply dapps/marketplace/src/pages/listings/_Search.js]: Sure

File: dapps/marketplace/src/pages/listings/Listings.js (hunk 1 above)
Comment (shahthepro) [reply dapps/marketplace/src/pages/listings/Listings.js]: No, can remove that prop.

File: dapps/marketplace/src/pages/listings/_Search.js (hunk 2 above)
Comment (shahthepro) [reply dapps/marketplace/src/pages/listings/_Search.js]: Will give it a try

**Comment by nick** [comment]
- [ ] Clicking the category in the dropdown doesn't close it
- [ ] Spacing in Zeplin is different for both mobile + desktop

**Pull request merged**